src/query_backend.h

```diff
@@ -377,6 +377,13 @@ class QueryableBackend {
 class FullBackend: public QueryableBackend, public RecBackend {
  public:
   virtual ~FullBackend() {}
+
+  /// Closes the backend, if approriate.
+  virtual void Close() {}
+
+ protected:
+  /// Flag for whether the backend is closed or not.
+  bool closed_ = false;
```

Comment:
I'd rather that we continue to maintain the precedent of these as abstract, stateless interfaces.  Implementations should decide whether to be idempotent or how they want to handle closing in general.